gitea: non-PR issues break JSON decoding

When a non-PR issue is encountered during issue search this breaks JSON decoding, resulting in an error:

`[ERROR] Error searching for merged PR: Network request failed: error decoding response body`

In this case the `pull_request` property is present in the response, but `null` in Gitea 1.25.5.

The source of this behavior seems to be https://github.com/robgonnella/releasaurus/blob/a2be2c61a2f68ebf07213f8396276870360d8f6c/src/forge/gitea/types.rs#L43

--- a/src/forge/gitea.rs
+++ b/src/forge/gitea.rs
@@ -705,7 +705,12 @@ impl Forge for Gitea {
             let issues: Vec<GiteaIssue> = result.json().await?;
 
             for issue in issues.iter() {
-                if !issue.pull_request.merged {
+                let is_merged = issue
+                    .pull_request
+                    .as_ref()
+                    .map(|pr| pr.merged)
+                    .unwrap_or(false);
+                if !is_merged {
                     log::warn!(
                         "found unmerged closed pr {} with pending label: skipping",
                         issue.number

--- a/src/forge/gitea/types.rs
+++ b/src/forge/gitea/types.rs
@@ -40,7 +40,7 @@ pub struct GiteaIssuePr {
 #[derive(Debug, Deserialize)]
 pub struct GiteaIssue {
     pub number: u64,
-    pub pull_request: GiteaIssuePr,
+    pub pull_request: Option<GiteaIssuePr>,
 }
 
 #[derive(Debug, Serialize)]

